ts/Series/PackedBubble/PackedBubbleLayout.ts

```diff
@@ -199,62 +204,93 @@ class PackedBubbleLayout extends ReingoldFruchtermanLayout {
+            ],
+            k = layout.k,
+            resolveSplitSeries = this.resolveSplitSeries;
 
-        let force: number,
-            distanceR: number,
-            distanceXY: Record<string, number>;
+        for (const node of nodes) {
+            const hasSplitSeries = resolveSplitSeries(node),
+                nodeSeries = node.series,
+                fixedPosition = node.fixedPosition,
+                paddedNodeRadius = (
+                    (node.marker?.radius || 0) +
+                    bubblePadding
+                );
 
-        nodes.forEach((node): void => {
             node.degree = node.mass;
             node.neighbours = 0;
-            nodes.forEach((repNode): void => {
-                force = 0;
+
+            for (const repNode of nodes) {
                 if (
                     // Node cannot repulse itself:
                     node !== repNode &&
                     // Only close nodes affect each other:
 
                     // Not dragged:
-                    !node.fixedPosition &&
+                    !fixedPosition &&
                     (
-                        layout.options.seriesInteraction ||
-                        node.series === repNode.series
+                        seriesInteraction || nodeSeries === repNode.series
+                    ) &&
+
+                    // Avoiding collision of parentNodes and parented points
+                    !(
+                        nodeSeries === repNode.series &&
+                        hasSplitSeries &&
+                        repNode.isParentNode
                     )
                 ) {
-                    distanceXY = layout.getDistXY(node, repNode);
-                    distanceR = (
-                        layout.vectorLength(distanceXY) -
-                        (
-                            (node.marker as any).radius +
-                            (repNode.marker as any).radius +
-                            bubblePadding
-                        )
-                    );
+                    const distanceXY = layout.getDistXY(node, repNode),
+                        distanceR = (
+                            layout.vectorLength(distanceXY) -
+                            (
+                                paddedNodeRadius + (repNode.marker?.radius || 0)
+                            )
+                        );
+
+                    let forceTimesMass: number | undefined;
+
                     // TODO padding configurable
                     if (distanceR < 0) {
                         (node.degree as any) += 0.01;
-                        (node.neighbours as any)++;
-                        force = layout.repulsiveForce(
-                            -distanceR / Math.sqrt(node.neighbours as any),
-                            layout.k,
-                            node,
-                            repNode
+                        // Const neighbours = ++(node.neighbours as any);
```

Comment:
Can be removed, because now we are iterating this value one step below, right?

Reply:
Good catch lol!